solidity/core/contracts/AbacusConnectionManager.sol

```diff
@@ -155,7 +168,14 @@ contract AbacusConnectionManager is IAbacusConnectionManager, Ownable {
      */
     function _unenrollInbox(address _inbox) internal {
         uint32 _currentDomain = inboxToDomain[_inbox];
-        domainToInbox[_currentDomain] = address(0);
+        address[] storage inboxes = domainToInboxes[_currentDomain];
+        for (uint256 i = 0; i < inboxes.length; i = i.add(1)) {
```

Comment:
While I think it's likely that a connection manager probably won't have many inboxes for a domain and iterating is probably not the hugest deal, I still think it'd be nice to have O(1) deletion.

Because a remote controller may own this contract and could be sending asynchronous calls, I think passing in the index of the inbox to unenroll could be race-prone and annoying to deal with. Wydt of having `domainToInboxes` be a `mapping(uint32 => EnumerableSet.AddressSet)`? It'd mean that the flat gas cost of enrolling will increase, but it'd give us O(1) insertion/deletion which will be more predictable